src/Components/signUp/SignUp.js

```diff
@@ -0,0 +1,245 @@
+  const [repeatPasswordUser, setRepeatPasswordUser] = useState("");
+  const [messageSignUp, setMessageSignUp] = useState({
+    text: "Enter your detalls below.",
+    messageColor: { color: "rgb(150, 157, 166)" },
+  });
+  const userInformation = JSON.parse(localStorage.getItem("userInformation"));
+  const { classes } = styles();
+
+  useEffect(() => {
+    if (userInformation.isLoggedIn) {
+      setIsLoggedIn(true);
+    }
+  }, []);
+
+  const enterDashboardFunction = (e) => {
+    if (userInformation !== null && userInformation.email === emailUser) {
+      theUserExistsMessage();
+    } else if (passwordUser === repeatPasswordUser && passwordUser !== "") {
+      saveInformationUser();
+      setIsLoggedIn(true);
+    } else {
+      passWordNotMatchMessage();
+    }
+  };
+
+  const theUserExistsMessage = () => {
+    setMessageSignUp({
+      text: "The user already exists. Log In.",
+      messageColor: { color: "rgb(237, 85, 151)" },
+    });
+  };
+
+  const passWordNotMatchMessage = () => {
+    setMessageSignUp({
+      text: "Passwords do not match.",
+      messageColor: { color: "rgb(237, 85, 151)" },
+    });
+  };
+
+  const saveInformationUser = (name, lastName, email, password) => {
+    let informationUser = null;
+
+    informationUser = {
+      name: nameUser,
+      lastName: lastNameUser,
+      email: emailUser,
+      password: passwordUser,
+      isLoggedIn: true,
+    };
+
+    localStorage.setItem("userInformation", JSON.stringify(informationUser));
+  };
+
+  if (isLoggedIn) {
+    return <Navigate to="/dashboard" />;
+  } else if (logInAccount) {
+    return <Navigate to="/" />;
+  } else {
+    return (
+      <Fragment>
+        <div className="sign-up">
+          <Image />
+          <section className="section-sign-up">
+            <div className="login-container-nav">
+              <nav className="login-link">
+                <span>Already have an account?</span>
+                <span
+                  className="link-l"
+                  onClick={() => {
+                    setLogInAccount(true);
+                  }}
+                >
+                  SIGN IN
+                </span>
+              </nav>
+            </div>
+            <div className="title-and-message">
+              <h1 className="title-sign-up">Get started absolutely free.</h1>
+              <p className="message-sign-up" style={messageSignUp.messageColor}>
+                {messageSignUp.text}
+              </p>
+            </div>
+            <form className="form-sign-up">
+              <TextField
+                className={classes.textFieldStyle}
+                label="First name"
+                variant="standard"
+                type="text"
+                InputLabelProps={{ className: "textfield-label" }}
+                name="Name"
+                sx={{
+                  "& .MuiInputLabel-root": {},
+                  borderBottom: "1px solid rgb(179, 175, 177)",
+                }}
```

Comment:
lets create a variable to set this styles, and reuse them every where, something like:
```
textFieldBorderStyle = {
                  "& .MuiInputLabel-root": {},
                  borderBottom: "1px solid rgb(179, 175, 177)
                }
```
then on lines 111 , 128, 145 just pass the variables